src/io/iohk/scalanet/peergroup/TCPPeerGroup.scala

```diff
@@ -33,7 +36,8 @@ import scala.util.Success
   * @param codec a decco codec for reading writing messages to NIO ByteBuffer.
   * @tparam M the message type.
   */
-class TCPPeerGroup[M](val config: Config)(implicit codec: Codec[M]) extends TerminalPeerGroup[InetMultiAddress, M]() {
+class TCPPeerGroup[M](val config: Config)(implicit codec: Codec[M], bufferInstantiator: BufferInstantiator[ByteBuffer])
```

Comment:
I don't know what `BufferInstantiator` is, so that might be obvious, but is there any reason you're making it configurable in that PR?